app/controllers/api/v0/service_plans_controller.rb

```diff
@@ -19,19 +21,36 @@ def order
 
       private
 
-      def order_service_plan_and_update_task(task, service_plan)
-        task.context = service_plan.order(order_params.slice(:service_parameters, :provider_control_parameters))
-        task.status = "completed"
-        task.save!
+      def service_plan_id
+        params[:service_plan_id].to_i
```

Comment:
This needs to reuse `params_for_order` below.

Reply:
I realized I wasn't using this anyway, so I just got rid of it